### PR TITLE
8295058: test/langtools/tools/javac 116 test classes uses com.sun.tools.classfile library

### PR DESCRIPTION
**Pull request by xqoasis**
Modified 116 tests with new Class-File API, and migrate helper methods to test/langtools/tools/javac/annotations/typeAnnotations/classfile/CombinationsTargetTest3.java while keep using old Class-File API

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295058](https://bugs.openjdk.org/browse/JDK-8295058): test/langtools/tools/javac 116 test classes uses com.sun.tools.classfile library (**Sub-task** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14874/head:pull/14874` \
`$ git checkout pull/14874`

Update a local copy of the PR: \
`$ git checkout pull/14874` \
`$ git pull https://git.openjdk.org/jdk.git pull/14874/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14874`

View PR using the GUI difftool: \
`$ git pr show -t 14874`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14874.diff">https://git.openjdk.org/jdk/pull/14874.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14874#issuecomment-1658780061)

### CODE REVIEW

File: test/langtools/tools/javac/classfiles/attributes/Signature/EnumTest.java
Comment (asotona) [inline test/langtools/tools/javac/classfiles/attributes/Signature/EnumTest.java:1]: This is probably a mistake, I see content of EnumTest replaced with Driver class.

**Review by asotona — COMMENTED** [review]
This is great amount of work well done.

File: test/langtools/tools/javac/classfiles/attributes/Signature/EnumTest.java
Comment (xqoasis) [reply test/langtools/tools/javac/classfiles/attributes/Signature/EnumTest.java:1]: Modified in the latest commit

File: test/langtools/tools/javac/annotations/typeAnnotations/classfile/CombinationsTargetTest3.java [hunk 1]
```diff
@@ -537,4 +539,206 @@ String sourceString(String testname, String retentn, String annot2,
...
+    File compile(File f) {
+        List<String> options = new ArrayList<>(List.of("-g", f.getPath()));
+        int rc = com.sun.tools.javac.Main.compile(options.toArray(new String[0]));
+        if (rc != 0)
+            throw new Error("compilation failed. rc=" + rc);
+        String path = f.getPath();
+        return new File(path.substring(0, path.length() - 5) + ".class");
+    }
+
+    /************ Helper annotations counting methods ******************/
```
[the remark this answers is not in the document]
Comment (xqoasis) [reply test/langtools/tools/javac/annotations/typeAnnotations/classfile/CombinationsTargetTest3.java:549]: Modified in the latest commit. I keep extending the ClassfileTestHelper, and only migrate the Test methods.

File: test/langtools/tools/javac/annotations/typeAnnotations/classfile/TestAnonInnerClasses.java [hunk 2]
```diff
@@ -343,8 +280,8 @@ void test(String ttype, ClassFile cf, Method m, Field f, boolean visible) {
                        vtaActual,itaActual,vaActual,iaActual);
     }
 
-    public void run() {
-        ClassFile cf   = null;
+    public <T extends Attribute<T>>void run() {
+        ClassModel cm   = null;
```
Comment (turbanoff) [inline test/langtools/tools/javac/annotations/typeAnnotations/classfile/TestAnonInnerClasses.java:284]:
```suggestion
        ClassModel cm  = null;
```

**Review by asotona — APPROVED** [review]
It looks good.
I think it can go in if you sync PR title "116 tests classes"  with bug title "116 test classes".
Thanks for the work.

**Comment by turbanoff** [comment]
>Modified 12 of 226 test/langtools/tools/javac test classes to replace com.sun.tools.classfile library.

Is this is still actual number? I mean 12.

**Pull request closed without merging**